### ISSUE TITLE
vcpkg install blocks for minutes in Windows 19041

### ISSUE THREAD

**Issue by sylveon** [microsoft/vcpkg#10991]
**Describe the bug**
I just updated my machine to the Insider release preview ring and now vcpkg takes several minutes at `Building package $PACKAGE[core]:$CONFIG...` for no reason

**Environment**
- OS: Windows 10 19041.207
- Compiler: Visual Studio 16.6.0 Preview 3

**To Reproduce**
1. `./vcpkg install discord-rpc --debug` (using that package as an example but happens for every package)
2. See vcpkg block at `Building package discord-rpc[core]:x64-windows...` for several minutes
3. Once it unblocks, see the extremely long time it took in `cmd_execute_and_stream_data`

**Expected behavior**
Install proceeds without blocking.

**Failure logs** 
```
[DEBUG] Using vcpkg-root: T:\Projects\vcpkg
Computing installation plan...
[DEBUG] Found path: C:\Program Files\CMake\bin\cmake.exe
[DEBUG] CreateProcessW("C:\Program Files\CMake\bin\cmake.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    17972 us
[DEBUG] CreateProcessW("C:\Program Files\CMake\bin\cmake.exe"  -P "T:/Projects/vcpkg/buildtrees/0.vcpkg_tags.cmake")
[DEBUG] cmd_execute_and_stream_data() returned 0 after    13951 us
The following packages will be built and installed:
    discord-rpc[core]:x64-windows
[DEBUG] <abientries>
[DEBUG] rapidjson|e19f4592ed0c708ae9c7de38f0915fecadabbc47
[DEBUG] CONTROL|1c796d7b13fa5c12af6b823c6bf85dea2f1a0e54
[DEBUG] disable-downloading.patch|a9f8d114b7f2f8251d82b6b0d5c8300ab2bb9f23
[DEBUG] portfile.cmake|e5547e25c873f313baa36d157968429a912055f0
[DEBUG] cmake|3.17.1
[DEBUG] powershell|ab1effc926b000a6adc12198a1886514ec203621a53b0cd7ec1cd9a8225dccda7e857feaabcfba4004bea73129b986abaad777c4573f44e0af70411226ce08b0
[DEBUG] vcpkg_fixup_cmake_targets|dac334f3bf50b8262be92623168023aa565aa729
[DEBUG] post_build_checks|2
[DEBUG] triplet|3eab5c64d9cf628fc0aec277fde66dbdc69b8516
[DEBUG] features|core
[DEBUG] </abientries>
Starting package 1/1: discord-rpc:x64-windows
Building package discord-rpc[core]:x64-windows...
[DEBUG] CreateProcessW("C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -legacy -products * -format xml)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    15214 us
[DEBUG] Found path: C:\Program Files\Git\cmd\git.exe
[DEBUG] CreateProcessW("T:\Projects\vcpkg\downloads\tools\git-2.25.1-1-windows\mingw32\bin\git.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    12145 us
[DEBUG] CreateProcessW(cmd /c ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Auxiliary\Build\vcvarsall.bat" -vcvars_ver=14.2 amd64   2>&1 <NUL" & echo cdARN4xjKueKScMy9C6H& set)
[DEBUG] cmd_execute_and_stream_data() returned 0 after 201462459 us
...
```

**Additional context**
After injecting a debugger, it seems that the following line blocks:
https://github.com/microsoft/vcpkg/blob/b6cd6ca0ba71667e46610e3ecdf5de0cef043ef5/toolsrc/src/vcpkg/base/system.process.cpp#L407

Manually running `vcvarsall.bat` does not result in such long times.

**Comment by Neumann-A** [comment]
duplicate of #10468

**Comment by sylveon** [comment]
My bad, missed that.
